src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs

```diff
@@ -205,6 +205,15 @@ public async Task GetContentAsync_ErrorStatusCode_ExpectedExceptionThrown(bool w
             }
         }
 
+        [Fact]
+        [OuterLoop("Slow - Negative connection test")]
+        public async Task GetContentAsync_WhenCanNotConnect_ExceptionContainsHostInfo()
+        {
+            using var client = new HttpClient(new SocketsHttpHandler());
```

Comment:
In tests, the `HttpClient` is usually created via `CreateHttpClient()`, e.g.: `using HttpClient httpClient = CreateHttpClient();`. Is there any reason here to create it directly?

Reply:
Both questions depend on what we consider to be the SUT. I can't decide what is a good functional unit to test this feature against. `HttpClient` with default setup or `SocketsHttpHandler`?

I have no preference, want to go with existing practices.

Reply:
Well, similarly looking test is here: https://github.com/dotnet/runtime/blob/master/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.cs#L2667-L2677
But that's a shared test code which gets executed for `WinHttpHandler` as well.
Maybe it would be sufficient to the existing test something like:
```
if (!IsWinHttpHandler)
{
    Assert.Contains(expected, exception.Message);
}
```
Note that we have many such `if`s in the shared test code, so this wouldn't be a revolutionary idea.

Reply:
Since I don't think it's worth to cover all the cases `HttpClientHandlerTest` covers, decided to keep the test in `HttpClientTest`  for now.